core/sdk-core/src/main/java/software/amazon/awssdk/core/async/AsyncRequestBody.java

```diff
@@ -143,25 +143,33 @@ static AsyncRequestBody fromString(String string) {
     }
 
     /**
-     * Creates a {@link AsyncRequestBody} from a byte array. The contents of the byte array are copied so modifications to the
-     * original byte array are not reflected in the {@link AsyncRequestBody}.
+     * Creates a {@link AsyncRequestBody} from a byte array.
      *
      * @param bytes The bytes to send to the service.
      * @return AsyncRequestBody instance.
      */
     static AsyncRequestBody fromBytes(byte[] bytes) {
-        return new ByteArrayAsyncRequestBody(bytes, Mimetype.MIMETYPE_OCTET_STREAM);
+        return ByteBuffersAsyncRequestBody.from(bytes);
     }
 
     /**
-     * Creates a {@link AsyncRequestBody} from a {@link ByteBuffer}. Buffer contents are copied so any modifications
-     * made to the original {@link ByteBuffer} are not reflected in the {@link AsyncRequestBody}.
+     * Creates a {@link AsyncRequestBody} from a {@link ByteBuffer}.
      *
      * @param byteBuffer ByteBuffer to send to the service.
      * @return AsyncRequestBody instance.
      */
     static AsyncRequestBody fromByteBuffer(ByteBuffer byteBuffer) {
-        return fromBytes(BinaryUtils.copyAllBytesFrom(byteBuffer));
+        return ByteBuffersAsyncRequestBody.of(null, byteBuffer);
+    }
+
+    /**
+     * Creates a {@link AsyncRequestBody} from an array of {@link ByteBuffer}.
+     *
+     * @param byteBuffers ByteBuffer[] to send to the service.
+     * @return AsyncRequestBody instance.
+     */
+    static AsyncRequestBody fromByteBuffers(ByteBuffer... byteBuffers) {
```

Comment:
Can we call this `fromByteBuffersUnsafe`?

A non-blocking observation: This method makes an interesting point, that we don't have a way of joining AsyncRequestBody implementations. This makes it work for byte buffers, but strings, byte arrays, etc. still need to be concatenated.

I almost wonder if we should have a `AsyncRequestBody append(AsyncRequestBody)` method on `AsyncRequestBody` that appends the data from one body to another, so that we don't have to have special treatment of byte buffers here.

`AsyncRequestBody.fromByteBuffer(firstBuffer).append(AsyncRequestBody.fromByteBuffer(secondBuffer));`

I don't think that would obviate the need for a simple way of joining multiple, like we have here, though. If you had 10 byte buffers you wanted to join, that seems like a pain.

Again, I don't expect you to change anything here for this PR... just documenting an observation.

Reply:
fyi, I added both the safe and unsafe implementation for this in the PR